pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,138 @@
+# size returns the size of an attribute
+let size = \length
+    cond {
+        length = 'MAX': '',
+        //seq.has_prefix('0x', length): '(' ++ //seq.trim_prefix('0x', length) ++ ')',
+        length > 0: $`(${length})`,
+        _: '',
+    };
+
+# attributePatternToken returns the token based on attribute value
+let attributePatternToken = \entity \attr \token
+    cond {
+        (attr.options?:{}): token,
+        (entity.primary_key where //seq.contains(attr.name, .)): token,
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+            .reference_attribute orderby .)) where . != {}; (fks): token,
+        (attr.length = 'MAX'): token,
+        (//seq.has_prefix('0x', attr.length)): token,
+        _: '',
+    };
+
+# entityPatternToken returns the token based on entity value
+let entityPatternToken = \entity \model \token
+    cond {
+        (entity.primary_key | entity.foreign_keys | entity.cluster | model.indexes): token,
+        _: '',
+    };
+
+# sortingOrder determines and appends sorting order
+let sortingOrder = \e \attr
+    let pk = //seq.split('(', //seq.join('', (e.primary_key where //seq.contains(attr.name, .)) orderby .));
+    cond {
+        pk count > 1: ', ~' ++ //seq.trim_suffix(')', //str.lower(pk(1))),
+        _: '',
+    };
+
+# compareColumnOrder compares the col order between primary key and table
+let compareColumnOrder = \entity
+    let pk = (entity.primary_key orderby .) >> //seq.split('(', .) >> .(0);
+    cond {
+        pk & (entity.attributes >> .name): false,
+        _: true,
+    };
+
+# attributePattern generates the pattern constituents depending on the mode
+let attributePattern = \entity \attr \mode
+    cond {
+        (mode = 0 && (entity.primary_key where //seq.contains(attr.name, .))): '~pk' ++ sortingOrder(entity, attr),
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+            .reference_attribute orderby .)) where . != {}; (mode = 1 && fks): '~fk',
```

Comment:
Nit: Having such a large cond expr is hard to read. it's better to move the heavy lifting into the branch like this:
```suggestion
        (mode = 1): 
            let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
                .reference_attribute orderby .)) where .;
                cond {fks:  '~fk'}
```

Reply:
ok